File: spring-transaction-test/src/main/java/com/gradle/sample/member/MemberApi.java
package com.gradle.sample.member;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/members")
public class MemberApi {

  @GetMapping("/{id}")
  public Member getMember(@PathVariable long id) {

    return Member.builder()
        .id(id)
        .email("yun@test.com")
        .name("name")
        .build();

  }

}

File: kotlin-jpa/src/main/kotlin/com/example/kotlinjpa/member/MemberApi.kt
package com.example.kotlinjpa.member

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/members")
class MemberApi(private val memberRepository: MemberRepository) {


    @GetMapping
    fun createMember(): Member {
        val member = Member(email = "asd@asd.com", name = "yun")
        return memberRepository.save(member)
    }

}